### ISSUE TITLE
Error inserting JSON data into '_aud' table after upgrading to Spring 3.1.1 and using @JdbcTypeCode(SqlTypes.JSON)

### ISSUE THREAD

**Issue by MarcinSzpigiel** [spring-projects/spring-data-jpa#3063]
After upgrading to the latest Spring 3.1.1, the @TypeDef annotation is no longer available. It has been replaced with the @JdbcTypeCode(SqlTypes.JSON) annotation. However, when persisting the entity, an error occurs while inserting into the _aud table:

Error Message:

`
could not execute batch [Batch entry 0 insert into user_aud (revision_type,id,revision_id,payload) values (0,1,152,'{}') was aborted: ERROR: column "payload" is of type jsonb but expression is of type character varying
Hint: You will need to rewrite or cast the expression.
`

Code Snippet

```
@Audited
@Entity
@Table(name = "user")
public class User {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
   @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private String payload;
    ...
    
```

**Comment by gregturn** [comment]
It would be better if you could include a full stack trace for the exception reported.

Also, this sounds very much like a Hibernate issue. I'd suggest creating a reproducer based purely on Hibernate and submitting an issue with them.

If you think this is actually a Spring Data issue, please provide a reproducer for the issue including a test case that demonstrates the correct behavior using the EntityManager directly with it's enver feature applied.

**Comment by MarcinSzpigiel** [comment]
Yes you are correct I think this is an issue in hibernate I raised that there. Thank you for your help @gregturn. Linked issue for the future reference https://hibernate.atlassian.net/browse/HHH-16910.
